packages/hardware-ledger/src/LedgerKeyAgent.ts

```diff
@@ -705,6 +705,37 @@ export class LedgerKeyAgent extends KeyAgentBase {
     return TransactionSigningMode.ORDINARY_TRANSACTION;
   }
 
+  /**
+   * There are currently two types of outputs supported by the ledger:
+   *
+   * legacy_transaction_output =
+   *   [ address
+   *   , amount : value
+   *   , ? datum_hash : $hash32
+   *   ]
+   *
+   * and
+   *
+   * post_alonzo_transaction_output =
+   *   { 0 : address
+   *   , 1 : value
+   *   , ? 2 : datum_option ; New; datum option
+   *   , ? 3 : script_ref   ; New; script reference
+   *   }
+   *
+   * Legacy outputs are definite length arrays of three elements, however the new babbage outputs are definite length maps
+   * of four elements.
+   *
+   * @param outputs The outputs to be verified.
+   */
+  hasBabbageOutput(outputs: Array<Serialization.TransactionOutput>): boolean {
```

Comment:
We can probably make this a method on the TransactionBody class

Reply:
Good idea, it would deduplicate this

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1616/commits/ab26ecf766b56801157ee80c28e75bb95e02c0ad and here https://github.com/input-output-hk/cardano-js-sdk/pull/1616/commits/b20f328f663180229f1e3931b923a448fde1bc49 and here https://github.com/input-output-hk/cardano-js-sdk/pull/1616/commits/2135507273da4890e0111c028dcf1db62a9dac5b